contentcuration/contentcuration/static/js/edit_channel/channel_set/views.js

```diff
@@ -39,12 +39,15 @@ var ChannelSetModalView = BaseViews.BaseView.extend({
         Vue.nextTick().then(this._mountVueComponent.bind(this));
     },
 
-    _handleSaving: function(saving, callback) {
+    _handleSaving: function(saving, close) {
         if(saving) {
             var self = this;
             store.dispatch('channel_set/saveChannelSet', function(model) {
                 self.options.onsave(model);
-                callback && callback();
+                if(close || store.state.channel_set.closing) {
+                  store.commit('channel_set/SET_CLOSING', false);
```

Comment:
I think I'd expect this kind of direct alteration of state to be hidden away in the Vuex action side of things, as much as possible.  Maybe using a separate `saveAndCloseChannelSet` action?

Reply:
I did consider that, but ultimately decided to set the state variable instead as the saveChannelSet is called from backbone, which made it more difficult to specify which save to call. Will re-examine once the channel list is in vue too though!